python/feathub/feature_views/feature_view.py

```diff
@@ -133,25 +133,35 @@ def get_output_fields(self, source_fields: List[str]) -> List[str]:
             raise FeathubException(
                 "Build this feature view before getting output fields."
             )
-        output_fields = []
+
+        feature_fields = [feature.name for feature in self.get_resolved_features()]
+
+        output_fields: List[str] = []
         if self.keep_source_fields:
-            output_fields.extend(source_fields)
-        elif self.timestamp_field is not None:
-            output_fields.append(self.timestamp_field)
-
-        for feature in self.get_resolved_features():
-            if feature.keys is not None:
-                output_fields.extend(feature.keys)
-            output_fields.append(feature.name)
-
-        # Order output fields similar to their order in the source table.
-        reordered_output_fields = []
-        for field in source_fields:
-            if field in output_fields:
-                reordered_output_fields.append(field)
-        reordered_output_fields.extend(output_fields)
-
-        return list(OrderedDict.fromkeys(reordered_output_fields))
+            source_fields_not_in_features = [
+                f for f in source_fields if f not in feature_fields
+            ]
+            output_fields.extend(source_fields_not_in_features)
+        else:
+            reserved_fields = (
+                [self.timestamp_field] if self.timestamp_field is not None else []
+            )
+            key_fields = [
+                key
+                for feature in self.get_resolved_features()
+                if feature.keys is not None
+                for key in feature.keys
+            ]
+            reserved_fields.extend(key_fields)
+            reserved_fields_not_in_features = [
+                f
+                for f in source_fields
+                if f in reserved_fields and f not in feature_fields
```

Comment:
It is supposed that all timestamp or key fields are source fields. Thus how about writing it this way?
```python
[f for f in reserved_fields if f not in feature_fields]
```

Reply:
Traversing in `source_fields` is to make the order of timestamp or key fields the same as those in `source_fields`.